src/Advection/vector_invariant_upwinding_treatment.jl

```diff
@@ -0,0 +1,148 @@
+                                
+Upwinding treatment for Divergence fluxes and Kinetic Energy gradient in the Vector Invariant formulation, whereas only 
+the terms corresponding to the transporting velocity are upwinded. (i.e., terms in `u` in the zonal momentum equation and 
+terms in `v` in the meridional momentum equation). Contrarily to `OnlySelfUpwinding`, the reconstruction (and hence the
+upwinding) is done _inside_ the gradient operator, i.e., velocities are reconstructed instead of velocity derivatives.
+
+Keyword arguments
+=================  
+
+- `cross_scheme`: Advection scheme used for cross-reconstructed terms (tangential velocities) 
+                    in the kinetic energy gradient and the divergence flux. Defaults to `CenteredSecondOrder()`.
+"""
+VelocityUpwinding(; cross_scheme = CenteredSecondOrder()) = VelocityUpwinding(cross_scheme)
+                    
+Base.summary(a::OnlySelfUpwinding)     = "OnlySelfUpwinding"
+Base.summary(a::CrossAndSelfUpwinding) = "CrossAndSelfUpwinding"
+Base.summary(a::VelocityUpwinding)     = "VelocityUpwinding"
+
+Base.show(io::IO, a::OnlySelfUpwinding) =
+    print(io, summary(a), " \n",
+            " KE gradient and Divergence flux cross terms reconstruction: ", "\n",
+            "    └── $(summary(a.cross_scheme))", "\n",
+            " Smoothness measures: ", "\n",
+            "    └── smoothness δU: $(a.δU_stencil)", "\n", 
+            "    └── smoothness δV: $(a.δV_stencil)", "\n",
+            "    └── smoothness δu²: $(a.δu²_stencil)", "\n",
+            "    └── smoothness δv²: $(a.δv²_stencil)")
+
+Adapt.adapt_structure(to, scheme::OnlySelfUpwinding) = 
+    OnlySelfUpwinding(Adapt.adapt(to, scheme.cross_scheme),
+                      Adapt.adapt(to, scheme.δU_stencil),
+                      Adapt.adapt(to, scheme.δV_stencil),
+                      Adapt.adapt(to, scheme.δu²_stencil),
+                      Adapt.adapt(to, scheme.δv²_stencil))
+
+Base.show(io::IO, a::CrossAndSelfUpwinding) =
+    print(io, summary(a), " \n",
+            " KE gradient cross terms reconstruction: ", "\n",
+            "    └── $(summary(a.cross_scheme))", "\n",
+            " Smoothness measures: ", "\n",
+            "    └── smoothness δ: $(a.divergence_stencil)", "\n", 
+            "    └── smoothness δu²: $(a.δu²_stencil)", "\n",
+            "    └── smoothness δv²: $(a.δv²_stencil)")
+
+Adapt.adapt_structure(to, scheme::CrossAndSelfUpwinding) = 
+    CrossAndSelfUpwinding(Adapt.adapt(to, scheme.cross_scheme),
+                          Adapt.adapt(to, scheme.divergence_stencil),
+                          Adapt.adapt(to, scheme.δu²_stencil),
+                          Adapt.adapt(to, scheme.δv²_stencil))
+
+
+Base.show(io::IO, a::VelocityUpwinding) =
+    print(io, summary(a), " \n",
+            " KE gradient and Divergence flux cross terms reconstruction: ", "\n",
+            "    └── $(summary(a.cross_scheme))")
```

Comment:
```suggestion
            "KE gradient and Divergence flux cross terms reconstruction: ", "\n",
            "└── $(summary(a.cross_scheme))")
```

There are extra spaces here and there in this file compared to what we normally use for `show`